osxkeychain.go

```diff
@@ -137,37 +138,48 @@ func authenticationTypeToGo(authtype C.CFTypeRef) AuthenticationType {
 
 // Adds an Internet password to the user's default keychain.
 func AddInternetPassword(pass *InternetPassword) error {
+	// TODO: Check fields for UTF-8 encoding and size fitting in
+	// 32 bits.
+
+	serverName := C.CString(pass.ServerName)
+	defer C.free(unsafe.Pointer(serverName))
+
+	// TODO: Make optional.
```

Comment:
how can this be made optional? Do we simply just pass a nil/null into the keychain API call? If so, can we do this?

```go
var securityDomain *C.char
if pass.SecurityDomain != "" {
	securityDomain = C.CString(pass.SecurityDomain)
	defer C.free(unsafe.Pointer(securityDomain))
}
```

Would there be any valid reason we'd want to support using an empty string in place of a variable like SecurityDomain?